enterprise/views.py

```diff
@@ -375,32 +465,71 @@ def get(self, request):
         course_id = request.GET.get('course_id', '')
         program_uuid = request.GET.get('program_uuid', '')
         self.preview_mode = bool(request.GET.get('preview_mode', False))
-        if not (enterprise_customer_uuid and success_url and failure_url):
-            raise Http404
 
-        if not self.preview_mode:
-            if not self.course_or_program_exist(course_id, program_uuid):
-                raise Http404
+        errored = False
+        enterprise_customer = None
+        if not (enterprise_customer_uuid and success_url and failure_url):
+            error_code = 'ENTGDS001'
+            LOGGER.error(
+                'Error: one or more of the following values was falsy: '
+                'enterprise_customer_uuid: {enterprise_customer_uuid}, '
+                'success_url: {success_url}, '
+                'failure_url: {failure_url}. '
+                'The following error code was reported to the user: {error_code}'.format(
+                    enterprise_customer_uuid=enterprise_customer_uuid,
+                    success_url=success_url,
+                    failure_url=failure_url,
+                    error_code=error_code,
+                )
+            )
+            messages.add_generic_error_message_with_code(request, error_code)
+            errored = True
```

Comment:
the reason this was the exception to the rule of rendering 404 immediately after an error was because i did not have the context_data yet, which requires an enterprise_customer to get the global context, so I deferred to waiting a little further down the code